src/tree/taffy_tree/tree.rs

```diff
@@ -420,9 +427,27 @@ impl Taffy {
         Ok(self.nodes[node.into()].cache.is_empty())
     }
 
+    /// Updates the stored layout of the provided `node` and its children
+    pub fn compute_layout_with_context(
+        &mut self,
+        node: NodeId,
+        available_space: Size<AvailableSpace>,
+        context: Measure::Context,
+    ) -> Result<(), TaffyError> {
+        self.context = Some(context);
+        let result = compute_layout(self, node, available_space);
+        self.context = None;
+        result
+    }
```

Comment:
I feel like this function could cause unintended side-effects because the `self.context` is overridden.
Maybe it could make sense to save the old `self.context` in a variable and then restore that instead of overwriting with `None`?

In any case, I think the behavior with `self.context` should be documented in the comment.

Reply:
That is the same conclusion I had when I experimented with this idea, so that's why I was super excited when your approach allowed to preserve the same API surface area (more or less). 

But I think fundamentally there needs to be another implementation of `LayoutTree` for more a "advanced" use case.